moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1981,10 +2055,58 @@ double moveit::core::RobotState::testJointSpaceJump(const JointModelGroup* group
   return percentage;
 }
 
+double moveit::core::RobotState::testJointSpaceJump(const JointModelGroup* group, std::vector<RobotStatePtr>& traj,
+                                                    double revolute_jump_threshold, double prismatic_jump_threshold)
+{
+  double percentage = 1.0;
+  bool still_valid = true;
+  const std::vector<const JointModel*>& joints = group->getActiveJointModels();
+  for (std::size_t i; i < traj.size() - 1 && still_valid; ++i)
+  {
+    for (std::size_t j = 0; j < joints.size() && still_valid; ++j)
+    {
+      const int idx = joints[j]->getFirstVariableIndex();
+      double dist = joints[j]->distance(traj[i]->position_ + idx, traj[i + 1]->position_ + idx);
+      if (joints[j]->getType() == JointModel::PRISMATIC)
+      {
+        // This is a prismatic joint
+        if (prismatic_jump_threshold > 0.0 && dist > prismatic_jump_threshold)
```

Comment:
combine this if with the above if statement (i think this is a speed up, not 100%)
same for REVOLUTE

Reply:
I separated them so I could use `else` to find any other JointType. I think this is optimal

Reply:
good point. but passing in the wrong joint type should never happen, so check that first in the for loop and throw an error

then you can combine these nested if statements